src/Reflex/Dom/SemanticUI/Dropdown.hs

```diff
@@ -283,3 +269,168 @@ semUiDropdownWithItems elId opts iv vals attrs = do
+  js_activateDropdown :: DOM.Element -> Callback (JSVal -> JSM ()) -> JSM ()
+#else
+activateDropdown :: DOM.Element -> (Text -> JSM ()) -> JSM ()
+activateDropdown e onChange = do
+  o <- obj
+  o ^. jss ("forceSelection"::Text) False
+  o ^. jss ("onChange"::Text) (fun $ \_ _ [t, _, _] ->
+    onChange =<< fromJSValUnchecked t)
+  o ^. jss ("fullTextSearch"::Text) True
+  void $ jsg1 ("$"::Text) e ^. js1 ("dropdown"::Text) o
+#endif
+
+#ifdef ghcjs_HOST_OS
+dropdownSetExactly :: DOM.Element -> [Int] -> JSM ()
+dropdownSetExactly e is = do
+  jsVal <- toJSVal $ map (tshow) is
+  js_dropdownSetExactly e jsVal
+
+foreign import javascript unsafe
+  "$($1).dropdown('set exactly', $2);"
+  js_dropdownSetExactly :: DOM.Element -> JSVal -> JSM ()
+#else
+dropdownSetExactly :: DOM.Element -> [Int] -> JSM ()
+dropdownSetExactly e is =
+  void $ jsg1 ("$"::Text) e ^. js2 ("dropdown"::Text) ("set exactly"::Text) is
+#endif
+
+------------------------------------------------------------------------------
+
+-- | Config for new semantic dropdowns
+data DropdownConf t a = DropdownConf
+  { _dropdownConf_setValue :: Dynamic t a
+  , _dropdownConf_attributes :: Map Text Text
+  , _dropdownConf_placeholder :: Text
+  }
+
+$(makeLenses ''DropdownConf)
+
+instance (Reflex t) => Default (DropdownConf t (Maybe a)) where
+  def = DropdownConf
+    { _dropdownConf_setValue = constDyn Nothing
+    , _dropdownConf_attributes = mempty
+    , _dropdownConf_placeholder = mempty
+    }
+
+instance (Reflex t) => Default (DropdownConf t [a]) where
+  def = DropdownConf
+    { _dropdownConf_setValue = constDyn mempty
+    , _dropdownConf_attributes = mempty
+    , _dropdownConf_placeholder = mempty
+    }
+
+instance HasAttributes (DropdownConf t a) where
+  type Attrs (DropdownConf t a) = Map Text Text
+  attributes = dropdownConf_attributes
+
+instance HasSetValue (DropdownConf t a) where
+  type SetValue (DropdownConf t a) = Dynamic t a
+  setValue = dropdownConf_setValue
+
+-- | Helper function
+indexToItem :: Vector (a, DropdownItemConfig m) -> Text -> Maybe a
+indexToItem items i' = do
+  i <- readMaybe $ T.unpack i'
+  fst <$> items !? i
+
+-- | Semantic-UI dropdown with static items
+semUiDropdownNew
+  :: (MonadWidget t m, Eq a)
+  => Vector (a, DropdownItemConfig m) -- ^ Items
+  -> [DropdownOptFlag]                -- ^ Options
+  -> DropdownConf t (Maybe a)         -- ^ Dropdown config
+  -> m (Dynamic t (Maybe a))
+semUiDropdownNew items options config = do
+  (el, evt) <- dropdownInternal items options False
+    (_dropdownConf_placeholder config) (_dropdownConf_attributes config)
+  let getIndex v = V.findIndex ((==) v . fst) items
+      setDropdown = dropdownSetExactly (_element_raw el)
+
+  -- setValue events
+  performEvent_ $ liftJSM . setDropdown . maybeToList . (>>= getIndex)
+               <$> updated (_dropdownConf_setValue config)
+
+  -- Set initial value
+  initialValue <- sample $ current $ _dropdownConf_setValue config
```

Comment:
A good rule of thumb with Reflex is to try to avoid calling `sample`.  There are almost always better ways to do things.  The above suggestion about splitting `setValue` into two fields will allow you to do this.